drop-transfer/src/storage_dispatch.rs

```diff
@@ -54,10 +54,7 @@ impl<'a> StorageDispatch<'a> {
                     .insert_outgoing_path_completed_state(transfer.id(), file_id.as_ref())
                     .await
             }
-            crate::Event::IncomingTransferCanceled(transfer, by_peer) => {
-                self.storage
-                    .insert_transfer_cancel_state(transfer.id(), *by_peer)
-                    .await;
+            crate::Event::IncomingTransferCanceled(transfer, _) => {
```

Comment:
This `insert_transfer_cancel_state` is removed in this case?

Reply:
Yes, since it was based on the event. The event is suppressed in some cases so no state would be inserted.
To fix this I needed to insert the state independently of the event